### PR TITLE
INGM-396 Add functions to scan drives with information

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

This PR adds methods to scan drives in ethercat and canopen including drive information (product code and revision number)

Fixes # INGM-396

### Type of change

Please add a description and delete options that are not relevant.

- [x] Add `scan_servos_ethercat_with_info`
- [x] Add `scan_servos_canopen_with_info`

### Tests
- [x] Add new unit tests if it applies.
- [ ] Run tests.

Please describe the tests that you ran to verify your changes if it applies. 

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] USe [type hints](https://docs.python.org/3/library/typing.html) for every function and argument.
- [x] Build documentation locally to verify changes.
- [ ] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingeniamotion tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

**Review by WoodElv — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - Add in configuration set_velocity_pid, set_position_pid and get_rated_current
 - Add functions to connect to and scan EtherCAT devices using CoE (SDOs).
 - Optional backup registers for Wizard tests.
+- New methods to scan the network and obtained drive info (product code and revision number).
```
Comment (MartinALL91) [inline CHANGELOG.md:26]: Obtain

File: ingeniamotion/communication.py [hunk 2]
```diff
@@ -646,11 +667,8 @@ def scan_servos_ethercat(
         Raises:
             TypeError: If some parameter has a wrong type.
         """
-        net = EthercatNetwork(interface_name)
-        slaves = net.scan_slaves()
-        if not isinstance(slaves, List):
-            raise TypeError("Slaves are not saved in a List")
-        return slaves
+        slaves_info = self.scan_servos_ethercat_with_info(interface_name)
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:670]: I think that changing this function is unnecessary.

File: ingeniamotion/communication.py [hunk 3]
```diff
@@ -646,11 +667,8 @@ def scan_servos_ethercat(
...
+        return list(slaves_info.keys())
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:671]: No need to use the `keys` method

File: ingeniamotion/communication.py [hunk 4]
```diff
@@ -709,10 +730,30 @@ def scan_servos_canopen(
                 baudrate,
             )
             return []
-        slaves = net.scan_slaves()
-        if not isinstance(slaves, List):
-            raise TypeError("Slaves are not saved in a List")
-        return slaves
+        slaves_info = net.scan_slaves_info()
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:733]: I think that changing this function is unnecessary.

**Review by GullinBustin — APPROVED** [review]

File: ingeniamotion/communication.py [hunk 5]
```diff
@@ -632,7 +633,27 @@ def connect_servo_ethercat_interface_ip(
...
+
+        Raises:
+            TypeError: If some parameter has a wrong type.
+        """
+        net = EthercatNetwork(interface_name)
+        slaves_info = net.scan_slaves_info()
+        return slaves_info
+
     def scan_servos_ethercat(
+        self,
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:656]: This method can be a static one.

**Pull request merged**